src/Controller/HomeController.php

```diff
@@ -0,0 +1,304 @@
+use App\Services\Registration;
+use App\Services\ForgotPassword;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+
+/**
+ * HomeController, this controller is to manage the register user through
+ * Registration class  and handles the login process also , On sucessful login
+ * this controller redirects to DashboardController.
+ *
+ *   @author rajdip <rajdip.roy@innoraft.com>
+ */
+class HomeController extends AbstractController
+{
+
+  /**
+   * It stores a object of EntityManagerInterface class
+   * It is to manage persistance and retriveal Entity object from Database.
+   *
+   *   @var object
+   */
+  public $em;
+
+  /**
+   * It stores a object of UserRepository class, it is to fetch data
+   * from user table of database.
+   *
+   *   @var object
+   */
+  public $userRepo;
+
+  /**
+   * It stores a object of User Class to set and get data from database
+   * through doctrine.
+   *
+   *   @var object
+   */
+  public $user;
+
+  /**
+   * It stores a object of Registration Class to validate the user input data
+   * at the time of registration.
+   *
+   *   @var object
+   */
+  public $registration;
+
+  /**
+   * It stores a object of ForgotPassword Class to validate user id, send otp,
+   * verify otp.
+   *
+   *   @var object
+   */
+  public $forgotPassword;
+
+  /**
+   *
+   * This constructor initializes object of HomeController Class also provides
+   * access to EntityManagerInterface object .
+   *
+   *   @param  object $em
+   *     It is to manage persistance and retriveal Entity object from Database.
+   *
+   *   @param object $session
+   *
+   *   @return void
+   *     Constructor returns nothing .
+   */
+  public function __construct(EntityManagerInterface $em)
+  {
+    $this->em = $em;
+    $this->user = new User();
+    $this->registration = new Registration($this->em);
+    $this->forgotPassword = new ForgotPassword();
+    $this->userRepo = $this->em->getRepository(User::class);
+  }
+
+  /**
+   * This method to take user credentials and checks in database whether those
+   * are valid or not , if the credentials are valid then send the user to the
+   * dashboard page otherwise return to the login page with error message.
+   *
+   *   @Route("/")
+   *     This route take user to the first page of website and then send user to
+   *     dashboard based on given credentials.
+   *
+   *   @param object $rq
+   *     This Request object is to handles the user credentials.
+   *
+   *   @return Response
+   *     Based on the validation the response is to either dashboard page or
+   *     the login page with the proper error messages.
+   */
+  public function login(Request $rq): Response
```

Comment:
Here after login also user can access login page.
Please add some condition like you add for dashboard page.